spec/scallop_spec.rb

```diff
@@ -99,6 +99,11 @@
       expect(result.output).to include('No such file or directory')
       expect(result.success?).to eq false
     end
+
+    specify 'working directory specified' do
+      expect(Open3).to receive(:capture3).with("ls -l", chdir: "/some/path")
+      Scallop.cmd(:ls, "-l").run(chdir: "/some/path")
```

Comment:
I think it would be way nicer and also much more in line with the spirit of immutability if chdir was another method on the builder 

so `Scallop.cmd(:ls, "-l").chdir("/some/path")` or potentiall even `cwd` instead of `chdir` 

WDYT?

Reply:
@reneklacan That was my initial thought; the reason I didn't go that route was that `Executor` receives a string and not a `CommandBuilder` instance where the `chdir` argument needs to be passed separate to `Open3`.

This would lead to a situation where you could create a parameter named `chdir` that would not expand into the actual command string, but be passed to Open3 instead.  

Although it might be worth it to think this through as we could also use other Open3 parameters (like `stdin_data`) in the future.

Reply:
ah, good point, let's do it this way then 👍